modules/git/repo_base_gogit.go

```diff
@@ -41,8 +41,14 @@ func openRepositoryWithDefaultContext(repoPath string) (*Repository, error) {
 	return OpenRepository(DefaultContext, repoPath)
 }
 
-// OpenRepository opens the repository at the given path within the context.Context
+// OpenRepository opens the repository at the given path within the context.Context.
 func OpenRepository(ctx context.Context, repoPath string) (*Repository, error) {
+	return OpenRepositoryWithAlternates(ctx, repoPath, "")
+}
+
+// OpenRepositoryWithAlternates opens the repository at the given path within the context.Context, with a provided alternate path.
+// altPath is ignored if empty.
+func OpenRepositoryWithAlternates(ctx context.Context, repoPath string, altPath string) (*Repository, error) {
```

Comment:
`altPath string` seems over-engineering

I think the code below is simple and good enough.

```
func OpenRepository(ctx context.Context, repoPath string) (*Repository, error) {
    ...
    options.AlternatesFS = osfs.New(repoPath)
}
```

Reply:
Because sometimes, it needs reference objects across git repositories. This is useful to reduplicate objects between forked repositories and parent repositories.

Reply:
I see, then the old commit (harryzcy's) is right and the newly changed commit seems wrong?

Reply:
@harryzcy 's patch cannot compile because non gogit doesn't support open repository with alternatives, I updated this PR to automatically load alternatives from the file, but it seems it will still fail. I don't know the reason.